### ISSUE TITLE
About mAP result

### ISSUE THREAD

**Issue by xxyll** [loiccordone/object-detection-with-spiking-neural-networks#6]
Hello, I want to discuss with you about COCO mAP results. I downloaded a train dataset, a valid dataset and a test dataset respectively 佛for my experiment.
The default value of batchsize in your code is 64, but according to the actual capacity of GPU, I set up batchsize as 8 in train step and 16 in test step. My result of mAP after training is 0.056, and after testing is 0.016, whether such results depend greatly on batchsize?

**Comment by wozcoder** [comment]
#7 Hello @xxyll I have seen that you have created an issue on the EOFError. This is what I am having right now. Could you please explain how you have solved this problem?
Regards,

**Comment by loiccordone** [comment]
Hello @xxyll, sorry for the delay I was on holidays.
This kind of results is obviously not normal. To answer your question I have not tested with smaller batch size, my guess is that as your batch size is smaller than the one I used, you need to adjust the learning rate. You could try to divide the default learning rate by 10 for example.

If it helps, here is my training loss and val mAP with default parameters and VGG-11 backbone. You can stop your training quickly if your loss doesn't drop below 1 before epoch 20.

![loss](https://user-images.githubusercontent.com/19710140/180188980-036bb1c2-ea37-4148-83d1-f9b92b1cad65.jpeg)

![val_AP_IoU= 5_ 05_ 95](https://user-images.githubusercontent.com/19710140/180188966-2683d16a-8612-4812-a60d-a9b58e7f2971.jpeg)

**Comment by xxyll** [comment]
Hi, I changed the learning rate to 0.0003, batchsize is also 8. After training with 50 epochs, the loss is 0.734, and the mAp is 0.141, but after testing the mAp result is 0.010. Does this mean that the change of learning rates have not worked?
I only downloaded train_a, val_a, test_a datasets from the official website for experiment, does the amount of datasets also affect the results?

**Comment by loiccordone** [comment]
I can't answer with certainty, but if your training converges (which seems to be the case) and the test mAP is bad, it means the network slightly overfitted your train dataset. In my opinion, it means you don't have enough train data. Try using more training samples. The learning seems fine.

**Comment by xxyll** [comment]
Hi, I find a problem I hadn't noticed before. After training, there will be generate a folder ‘ckpt-od-gen1-vgg-11’, and there are three '.ckpt' files in it. 
After training the mAp is 1.162, the loss of epoch49 is 0.744. But the files in ‘ckpt-od-gen1-vgg-11’ is 'gen1-epoch=00-train_loss=9.3256.ckpt', 'gen1-epoch=01-train_loss=2.2496.ckpt', 'gen1-epoch=02-train_loss=1.7713.ckpt'. It seems that only the results of the first three rounds have been saved.
Is the pretrained model used one of the three files in the test step？
